src/txn/txn_timestamp.c

```diff
@@ -441,6 +441,19 @@ __wt_txn_global_set_timestamp(WT_SESSION_IMPL *session, const char *cfg[])
     if (has_oldest || has_stable)
         __wt_txn_update_pinned_timestamp(session, force);
 
+    /*
+     * Even if the timestamps have been forcibly set, they must always satisfy the condition that
+     * oldest <= stable. Don't fail as MongoDB violates this rule in very specific scenarios.
+     */
+    if (txn_global->has_stable_timestamp && txn_global->has_oldest_timestamp &&
+      txn_global->oldest_timestamp > txn_global->stable_timestamp) {
+        WT_STAT_CONN_INCR(session, txn_set_ts_out_of_order);
+        __wt_verbose_warning(session, WT_VERB_TIMESTAMP,
```

Comment:
Since we know that MongoDB hits this path, this log will start showing up. Should we still keep it?

Reply:
I think you should check this before we release the write lock.

Reply:
That's a very good point about the write lock, I missed that, thank you.

Reply:
Fixed in 5be0e2e5ff9724d9091ad336720fdfd44bde3d1f.

Reply:
I haven't kept up with how these map to MongoDB logging, do you know if this is enabled with the default log levels?
I think it depends on how often the message appears, as to whether it should be included.
If it only occurs in MongoDB in exceptional circumstances (e.g: replication failover due to a primary crash), then having this generated always is fine. If it will be generated during a "normal" startup process, it'd be better to only enable it when more verbose logging is enabled by the user.

Reply:
This does not seem to only occur during exceptional circumstances given SERVER-84706. Using `__wt_verbose_warning` would make the log appear every time the server does it (SERVER-84706 will clarify this). The solution here is what you said and use `__wt_verbose_debug1` so it appears only if the server has configured verbosity for `WT_VERB_TIMESTAMP`.

Reply:
See 2179d6b113268a253e54c05312b47d3c31dfe8e9.